### PR TITLE
Add aad-only auth template

### PR DESCRIPTION
**Pull request by VanMSFT**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        },
+        "aad_admin_objectid": {
+            "type": "String",
+            "metadata": {
+                "description": "The Object ID of the Azure AD admin."
+            }
+        },
+        "aad_admin_tenantid": {
+            "type": "String",
+            "metadata": {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json:33]: add a defaultValue of "[subscription().tenantId]" and remove the parameter value from the parameters file

File: quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            "defaultValue": "User",
+            "allowedValues": [
+                "User",
+                "Group",
+                "Application"
+            ],
+            "type": "String"
+        },
+        "aad_only_auth": {
+            "defaultValue": true,
```
Comment (bmoore-msft) [inline quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json:47]: what happens if this is set to false?  does the deployment fail or is the server created with no way to log into it?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json (hunk 1 above)
Comment (VanMSFT) [reply quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json:33]: Thanks! I'll make this change

File: quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json (hunk 2 above)
Comment (VanMSFT) [reply quickstarts/microsoft.sql/sql-logical-server-aad-only-auth/azuredeploy.json:47]: The deployment should not fail. This feature only disables SQL Authentication, but the Azure AD user that was added should still be able to log in.

**Pull request merged**